functions.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+function toJson(array $data): ?string
+{
+    try {
+        $jsonEncoded = json_encode($data, JSON_ERROR_SYNTAX, 512);
+    } catch (JsonException $e) {
+        echo $e->getMessage();
+        return null;
+    }
+
+    return $jsonEncoded;
+}
+
+function fromJson(string $data): ?array
+{
+    try {
+        $jsonDecoded = json_decode($data, true, 512, JSON_ERROR_SYNTAX);
+    } catch (JsonException $e) {
+        echo $e->getMessage();
```

Comment:
Errors should be logged in a file or in stderr. Usage of `echo` will return error to the client as a response, if it is called during request handling.

Reply:
@AndreiOrmanji - Function added which logs errors to the file